src/NServiceBus.Persistence.DynamoDB/Serialization/HashSetStringConverter.cs

```diff
@@ -0,0 +1,109 @@
+                binder: null,
+                args: null,
+                culture: null)!;
+            return converter;
+        }
+
+        sealed class SetConverter<TSet> : JsonConverter<TSet> where TSet : ISet<string>
+        {
+            public override TSet? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+            {
+                if (reader.TokenType != JsonTokenType.StartObject)
+                {
+                    throw new JsonException();
+                }
+
+                reader.Read();
+                if (reader.TokenType != JsonTokenType.PropertyName)
+                {
+                    throw new JsonException();
+                }
+
+                string? propertyName = reader.GetString();
+                if (propertyName != PropertyName)
+                {
+                    throw new JsonException();
+                }
+
+                reader.Read();
+                if (reader.TokenType != JsonTokenType.StartArray)
+                {
+                    throw new JsonException();
+                }
+
+                // Deliberately not passing the options to use the default json serialization behavior
+                var set = JsonSerializer.Deserialize<TSet>(ref reader);
+
+                reader.Read();
+
+                if (reader.TokenType != JsonTokenType.EndObject)
+                {
+                    throw new JsonException();
+                }
+                return set;
+            }
+
+            public override void Write(Utf8JsonWriter writer, TSet value, JsonSerializerOptions options)
+            {
+                writer.WriteStartObject();
+                writer.WritePropertyName(PropertyName);
+                // Deliberately not passing the options to use the default json serialization behavior
+                JsonSerializer.Serialize(writer, value);
+                writer.WriteEndObject();
+            }
+        }
+
+        public static bool TryExtract(JsonProperty property, out List<string?>? strings)
+        {
+            strings = null;
+            if (!property.NameEquals(PropertyName))
+            {
+                return false;
+            }
+
+            foreach (var innerElement in property.Value.EnumerateArray())
+            {
+                strings ??= new List<string?>(property.Value.GetArrayLength());
+                strings.Add(innerElement.GetString());
+            }
+
+            strings ??= new List<string?>(0);
+            return true;
+        }
+
+        public static JsonNode ToNode(List<string> strings)
+        {
+            var jsonObject = new JsonObject();
+            var stringHashSetContent = new JsonArray();
+            foreach (var stringValue in strings)
+            {
+                stringHashSetContent.Add(stringValue);
+            }
+            jsonObject.Add(PropertyName, stringHashSetContent);
+            return jsonObject;
```

Comment:
I kinda feel we don't need to go this path because we don't need to maintain the information about this needing to be a hashset, Json serializer can map arrays on it's own to HashSets, so I think this can be replaced by:

```
        public static JsonNode ToNode(List<string> strings)
        {
            var stringHashSetContent = new JsonArray();
            foreach (var stringValue in strings)
            {
                stringHashSetContent.Add(stringValue);
            }
            return stringHashSetContent;
        }
```

which will then simplify the `Read` operation to:

```
            public override TSet? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
            {
                // Deliberately not passing the options to use the default json serialization behavior
                var set = JsonSerializer.Deserialize<TSet>(ref reader);
                return set;
            }
```